addons/website_sale/models/sale_order.py

```diff
@@ -98,18 +98,34 @@ def _website_product_id_change(self, order_id, product_id, qty=0):
             'pricelist': order.pricelist_id.id,
         })
         product = self.env['product.product'].with_context(product_context).browse(product_id)
-        pu = product.price
-        if order.pricelist_id and order.partner_id:
-            order_line = order._cart_find_product_line(product.id)
-            if order_line:
-                pu = self.env['account.tax']._fix_tax_included_price_company(pu, product.taxes_id, order_line[0].tax_id, self.company_id)
+        discount = 0
+
+        if order.pricelist_id.discount_policy == 'without_discount':
+            # This part is pretty much a copy-paste of the method '_onchange_discount' of
```

Comment:
in this case, can we call this method instead to duplicate all this code ?

Reply:
There is always a way to do it.

In this case, note that the SO line might not exist yet:

https://github.com/odoo/odoo/blob/98fe5054c26828a3a15c8cd7b5905c7cf99c6103/addons/website_sale/models/sale_order.py#L169-L171

So we'd also have to refactor the onchange.